### DISCUSSION TITLE
请问如何优雅/快速/简洁的实现background-image

### DISCUSSION THREAD

**Discussion by LearningUnity** [unocss/unocss#3312] [Q&A] [answered]
目前我的方案是这样: 
![Snipaste_2023-11-06_15-09-37](https://github.com/unocss/unocss/assets/17720234/d0ca8f96-2eb9-426e-a8eb-fbacc3a6160c)
问题: 项目中的所有背景都会暴露在全局样式中
请问大佬们知道如何将 部分原子局部化 么, 只需要将背景css作用域修改为 scoped

**Comment by zyyv** [comment, marked as answer]
有内部规则实现 `background-image`， `bg-[url(图片地址)]`

>  [!WARNING]
> 仅适用于 网络图片，本地资源图片不会在打包时进行资源路径转换

**Reply by LearningUnity** [reply]
谢谢大佬回复~
![image](https://github.com/unocss/unocss/assets/17720234/372ff1fc-969c-4923-80d2-ac3e762ad777)
这种方式需要使用class, 请问大佬有不需要class直接使用属性的方式嘛, 能否帮忙举个例子啊; 
另外这种方式也是在全局样式中生成的, unocss有部分样式作用域为局部的设置么

**Reply by zyyv** [reply]
> 请问大佬有不需要class直接使用属性的方式嘛

使用 class 没什么不好，部分规则的使用格式情况不满足于 标签属性 的规范。当不满足时，只能使用 class
或者 将其 转换为 shortcuts 就可以使用属性模式了

> unocss有部分样式作用域为局部的设置么

可以参考使用 `scope` variant, https://unocss.dev/interactive/?s=scope-[.foo]:text-red

**Reply by LearningUnity** [reply]
感谢大佬, 我改成这样了, 有点像 img 标签了, 看起来还不错, 哈哈哈
![image](https://github.com/unocss/unocss/assets/17720234/2fc31ac9-7110-4cf1-b4fd-366b8a289374)

**Reply by zyyv** [reply]
> 仅适用于 网络图片，本地资源图片不会在打包时进行资源路径转换

应该打包有问题。。。 自己再看看吧
